src/redownload/web_parsing.py

```diff
@@ -0,0 +1,58 @@
+""" The module for HTML parsing related functions. """
```

Comment:
This code makes me so happy. Short functions, docstrings, good abstraction, type hints. Thank you.